sklearn/metrics/pairwise.py

```diff
@@ -585,6 +592,119 @@ def cosine_distances(X, Y=None):
     return S
 
 
+def gower_distances(X, Y=None, w=None, categorical_features=None):
+    """
+    Computes the gower distances between X and Y
+
+    Read more in the :ref:`User Guide <metrics>`.
```

Comment:
The user guide says nothing about this metric, and I think it ought to have a section there. Please describe Gower there, including its formula, its benefits and drawbacks, and the usage of weights.